DEV-11_Translit

nikitoshom opened the pull request:
Files of DEV-11: ConvertString.cs, Program.cs.

Commit:
DEV-11_Translit

Files of DEV-11: ConvertString.cs, Program.cs.

Commit:
Update Program.cs

IgorKheidorov commented on ConvertString.cs:
```diff
@@ -0,0 +1,59 @@
+   * This class take input string and conver from Cyrillica into Latin and vice versa.
+   * There we have 4 strings which include small and big letters of each language.
+   */
+  class ConvertString
```
StringConvertor

IgorKheidorov commented on ConvertString.cs:
```diff
@@ -0,0 +1,59 @@
+   */
+  class ConvertString
+  {
+    string[] lat_up = { "Kh", "Ts", "Shch", "Sh", "Ch", "A", "B", "V", "G", "D", "E", "Iy", "Ia", "Yu", "Ya", "Ye", "Ye", "Ye", "Ye", "Ye", "Ye", "Y", "\"", "'", "Zh", "Z", "I", "Y", "K", "L", "M", "N", "O", "P", "R", "S", "T", "U", "F"};
```
Code style violation, never use "_" in value names, use more clear name

IgorKheidorov replied:
seems that const string[] will be better

IgorKheidorov commented on ConvertString.cs:
```diff
@@ -0,0 +1,59 @@
+     * @return boolean LanguageEN (if LanguageEN == true, it means that our string was written in English, if LanguageEN == false, string was written in Russia)
+     */
+    public Boolean ChoseLanguage(string str) {
+      Boolean languageEN = false;
```
Bad idea, use enum for language marker, not bool. Imagine that there will be three languages. Bad name

IgorKheidorov commented on ConvertString.cs:
```diff
@@ -0,0 +1,59 @@
+     */
+    public Boolean ChoseLanguage(string str) {
+      Boolean languageEN = false;
+      for (int i = 0; i < lat_low.Length; i++)
```
what about one of the letters is not latin? No need to check the whole string, or make validator that all letters are latin or not

IgorKheidorov commented on ConvertString.cs:
```diff
@@ -0,0 +1,59 @@
+     * @param string
+     * @return convertString
+     */
+    public string Transliter(string str)
```
Use verb as methods names!!!!

IgorKheidorov commented on ConvertString.cs:
```diff
@@ -0,0 +1,59 @@
+      }
+      else
+      {
+        for (int i = 0; i < lat_low.Length; i++)
```
It seems that wrong condition!  lat_low.Length -> rus

IgorKheidorov commented on ConvertString.cs:
```diff
@@ -0,0 +1,59 @@
+      string convertString = str;
+      if (ChoseLanguage(str))
+      {
+        for (int i = 0; i < lat_low.Length; i++)
```
lat_low.Length is limit, are you sure that lat_up[i] will be in range?

IgorKheidorov commented on Program.cs:
```diff
@@ -0,0 +1,26 @@
+    {
+      try
+      {
+        ConvertString dictionary = new ConvertString();
```
ConvertString dictionary very bad name

Commit:
Update and rename ConvertString.cs to StringConvertor.cs

I could'n change "string[]" on "const string[]". Maybe at the lessons i will ask you and you will show how i can do it.

Commit:
Update Program.cs

IgorKheidorov commented on Program.cs:
```diff
@@ -0,0 +1,26 @@
+      {
+        StringConvertor stringConvertor = new StringConvertor();
+        string str = Console.ReadLine();
+        str = stringConvertor.Translaite(str);
```
Grammar

IgorKheidorov commented on StringConvertor.cs:
```diff
@@ -0,0 +1,76 @@
+    public int ChoseLanguage(string str) {
+      int language = (int)Language.EN;
+      int countEng = 0, countRus = 0;
+      for (int i = 0; i < numIter; i++)
```
bad idea, use one limit to all striings, numIter = 39 very bad, possibly OutOfRange exception

IgorKheidorov replied:
Use only massiveName.Length limit for cycles

IgorKheidorov commented on StringConvertor.cs:
```diff
@@ -0,0 +1,76 @@
+          countRus++;
+        }
+      }
+      if (countEng > countRus)
```
return (countEng > countRus) ? Language.EN : Language.RU

IgorKheidorov commented on StringConvertor.cs:
```diff
@@ -0,0 +1,76 @@
+     * @param string
+     * @return int language (If count of russian letters more than enland letters in string, then program will translate from russian and vice versa.)
+     */
+    public int ChoseLanguage(string str) {
```
Language  ChoseLanguage(string str)

IgorKheidorov replied:
Not ChoseLanguage, DetermineLanguage

IgorKheidorov commented on StringConvertor.cs:
```diff
@@ -0,0 +1,76 @@
+      string convertString = str;
+      switch (ChoseLanguage(str))
+      {
+        case 0:
```
use enum for comparison

IgorKheidorov commented on StringConvertor.cs:
```diff
@@ -0,0 +1,76 @@
+      switch (ChoseLanguage(str))
+      {
+        case 0:
+          for (int i = 0; i < numIter; i++)
```
engUp.Length as limit

IgorKheidorov reviewed (approved):
simple clear code, good, but what about non standard situations and rules as in requirements?
